pkg/apis/pipeline/v1alpha1/volume_resource_test.go

```diff
@@ -0,0 +1,158 @@
+/*
+ Copyright 2019 The Tekton Authors.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+     http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package v1alpha1_test
+
+import (
+	"github.com/google/go-cmp/cmp"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	tb "github.com/tektoncd/pipeline/test/builder"
+	"github.com/tektoncd/pipeline/test/names"
+	corev1 "k8s.io/api/core/v1"
+	"testing"
+)
+
+func TestNewVolumeResource(t *testing.T) {
+	for _, c := range []struct {
+		desc     string
+		resource *v1alpha1.PipelineResource
+		want     *v1alpha1.VolumeResource
+	}{{
+		desc: "basic volume resource",
+		resource: tb.PipelineResource("test-volume-resource", "default", tb.PipelineResourceSpec(
+			v1alpha1.PipelineResourceTypeVolume,
+			tb.PipelineResourceSpecParam("name", "test-volume-resource"),
+		)),
+		want: &v1alpha1.VolumeResource{
+			Name: "test-volume-resource",
+			Type: v1alpha1.PipelineResourceTypeVolume,
+			Size: "5Gi",
+		},
+	}, {
+		desc: "volume resource with size",
+		resource: tb.PipelineResource("test-volume-resource", "default", tb.PipelineResourceSpec(
+			v1alpha1.PipelineResourceTypeVolume,
+			tb.PipelineResourceSpecParam("name", "test-volume-resource"),
+			tb.PipelineResourceSpecParam("size", "10Gi"),
+		)),
+		want: &v1alpha1.VolumeResource{
+			Name: "test-volume-resource",
+			Type: v1alpha1.PipelineResourceTypeVolume,
+			Size: "10Gi",
+		},
+	}} {
+		t.Run(c.desc, func(t *testing.T) {
+			got, err := v1alpha1.NewVolumeResource(c.resource)
+			if err != nil {
+				t.Errorf("Test: %q; TestNewVolumeResource() error = %v", c.desc, err)
+			}
+			if d := cmp.Diff(got, c.want); d != "" {
+				t.Errorf("Diff:\n%s", d)
+			}
+		})
+	}
+}
+
+func Test_VolumeResource_GetDownloadContainerSpec(t *testing.T) {
+	names.TestingSeed()
+	testcases := []struct {
+		name           string
+		volumeResource *v1alpha1.VolumeResource
+		wantContainers []corev1.Container
+		wantErr        bool
```

Comment:
if possible i think tests end up a bit easier to read and update if instead of having  `wantErr` we create separate tests for error cases (and in this case we're not actually using `wantErr`)